lib/lesson8/aop/aspect_oriented.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:mirrors';
+
+import 'package:dart_learning_journey/lesson8/aop/task_annotation.dart';
+import 'package:dart_learning_journey/lesson8/logging/custom_logs.dart';
+import 'package:talker/talker.dart';
+
+class _TaskEntry {
+  final void Function(Talker logger) func;
+  final int order;
+
+  _TaskEntry(this.func, this.order);
+}
+
+class AspectOriented {
+  final Talker _logger;
+
+  AspectOriented(this._logger);
+
+  void run({required Object libraryAnchor}) {
+
+    // libraryAnchor is not used in this method but this parameter is mandatory because Dart has an interesting
+    // mechanism of three shaking, even aggressive I would say, so we need to make sure that code in which
+    // there may be tasks for execution loaded into memory.
+
+    _logger.info('Searching for @task functions using reflect...');
+
+    final mirrorSystem = currentMirrorSystem();
+    final ClassMirror talkerClassMirror = reflectClass(Talker);
+    final List<_TaskEntry> tasksToRun = [];
+
+    // Since it's just homeworks skipped advanced checks what 'main' here since we have more than one main.
+    final Map<Uri, LibraryMirror> currentLibraries = mirrorSystem.libraries;
+
+    for (final MapEntry<Uri, LibraryMirror> currentLibrary in currentLibraries.entries) {
+      _logger.debug('Inspecting library: ${currentLibrary.key.toString()}');
+
+      for (final DeclarationMirror declaration in currentLibrary.value.declarations.values) {
+        if (declaration is MethodMirror && declaration.isStatic ||
+            (declaration is MethodMirror && declaration.isTopLevel)) {
```

Comment:
```suggestion
       if (declaration case MethodMirror(isStatic: true) || MethodMirror(isTopLevel: true)) {
```
You can use patter matching here

Reply:
y, good point )